src/Slot.ts

```diff
@@ -191,13 +203,44 @@ export function connectSlot<T = void, T2 = void>(
         ) => {
             const paramHandlers = handlers[transportKey][param] || []
             const handlerIndex = paramHandlers.indexOf(handler)
-            if (handlerIndex > -1) handlers[transportKey][param].splice(handlerIndex, 1)
-            if (getParamHandlers(param, handlers).length === 0) callLazyDisonnectCallbacks(param)
-            awaitHandlerRegistration(String(transportKey), param)
+
+            if (handlerIndex > -1)
+                handlers[transportKey][param].splice(handlerIndex, 1)
+
+            if (getParamHandlers(param, handlers).length === 0)
+                callLazyDisonnectCallbacks(param)
+
+            if (remoteHandlersConnected[transportKey])
+                awaitHandlerRegistration(String(transportKey), param)
+
         }
 
-        transport.addRemoteHandlerRegistrationCallback(slotName, remoteHandlerRegistered)
-        transport.addRemoteHandlerUnregistrationCallback(slotName, remoteHandlerUnregistered)
+        const remoteEventListChangedListener = () => {
+            if (remoteHandlersConnected[transportKey]) {
+                Object.keys(remoteHandlersConnected[transportKey]).forEach(
+                    (param) => {
+                        remoteHandlersConnected[transportKey][
+                            param
+                        ].onRegister()
+                    }
+                )
+            }
+            delete remoteHandlersConnected[transportKey]
```

Comment:
So when creating an event bus between 2 ends, on each side handlers handlers will be registered for the remote end. In this case the remote end has communicated to this ends (local) that he does not want to listed to a specific event. Because the local hands has declared a handlers we need to resolve the promise (which is what we do with the onRegister). We need then to remove the handler from the list as the remote end does not listen to the event.